var/spack/repos/builtin/packages/mofem-users-modules/package.py

```diff
@@ -0,0 +1,97 @@
+##############################################################################
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+import os
+
+
+class MofemUsersModules(CMakePackage):
+    """mofem users modules"""
+
+    homepage = "http://mofem.eng.gla.ac.uk"
+    url = "https://bitbucket.org/likask/mofem-joseph/downloads/users_modules_dummy"
+    version('1.0', '5a8b22c9cdcad7bbad92b1590d55edb1', expand=False)
+
+    maintainers = ['likask']
+
+    variant('copy_user_modules', default=True,
+        description='Copy user modules directory instead linking')
+    variant('with_metaio', default=False,
+        description='Install MetaIO with MoFEM users modules')
+
+    variant('mofem_fracture_module', default=False,
+        description="Install fracture mechanics module")
+    variant('mofem_minimal_surface_equation', default=False,
+        description="Install minimal surface equation module")
+
+    extends('mofem-cephas')
+
+    depends_on("mofem-fracture-module", type=('build', 'link', 'run'),
+        when='+mofem_fracture_module')
+    depends_on("mofem-minimal-surface-equation", type=('build', 'link', 'run'),
+        when='+mofem_minimal_surface_equation')
+
+    @property
+    def root_cmakelists_dir(self):
+        """The relative path to the directory containing CMakeLists.txt
```

Comment:
maybe the default docstring and somewhere add a simple comment which explains what's going on for others (including reviewers) who might came across this package.
Something like (adjust as you wish):

> MofemUsersModules collects all user-provided modules and extend mofem-cephas package.
The CMakeList file for user modules is located in `mofem-cephas`. 
MofemUsersModules itself does not contain any code (is a dummy). 
We build in the self.prefix/build and provide sources location of user modules, i.e. `mofem-fracture-module`.
Those are kept as a stand-alone packages (instead of resources) as they have different versions and developers.
<describe-what-happens-next-as-i-don't-understand-relationship-with-mofem>
```

Reply:
Thanks very much for this. The part which you write describes what happens very well.